src/confirmation_strategies.rs

```diff
@@ -277,7 +280,7 @@ pub fn confirmations_by_blocks(
                 if block_slots.is_empty() {
                     continue;
                 }
-                start_block = *block_slots.last().unwrap();
+                start_block = *block_slots.last().unwrap() + 1;
```

Comment:
why is this one extended by 1?

Reply:
This is to solve the duplicate blocks issue. We have already got all blocks till here, next time we will fetch last_block + 1 till end.